src/layers/tile/mod.rs

```diff
@@ -107,6 +107,7 @@ impl TileLayerData {
 #[derive(Debug, Clone, PartialEq)]
 pub struct LayerTile<'map> {
     pub tileset: &'map Tileset,
+    pub tileset_index: usize,
```

Comment:
Another way of doing this would be to store the index in the tileset itself.
That would make the facade type LayerTile<'map> slightly cheaper to construct, though makes Tileset more tightly coupled with Map, which might be awkward.

This is my preferred way at the moment.

Reply:
Right, it's not possible to store the tileset index on the tileset, because a tileset can be used by multiple loaded maps, and it could have a different index in each of those maps.